src/libpmemfile-core/data.c

```diff
@@ -365,9 +381,9 @@ file_allocate_range(PMEMfilepool *pfp, struct pmemfile_vinode *vinode,
 
 			block->offset = offset;
 
-			block->next =
-			    (TOID(struct pmemfile_block))
-			    pmemobj_oid(vinode->first_block);
+			block->next = blockp_as_oid(vinode->first_block);
+			block->prev = TOID_NULL(struct pmemfile_block);
+			vinode->first_block->prev = blockp_as_oid(block);
```

Comment:
Then it means this code path is not exercised under pmemcheck. Please write a test case for this code path.